lib/authOptions.ts

```diff
@@ -59,27 +40,26 @@ export const authOptions: NextAuthOptions = {
     }),
     Credentials({
       name: "Credentials",
-      // TODO: im a little confused as to how the signin flow works, since a signUp needs to create a new user and the form has
-      // TODO: extra data (full name) and needs some way of flagging this afterr sign up to redirect to onboarding
+
       credentials: {
         username: { label: "Email", type: "email", placeholder: "..." },
         password: { label: "Password", type: "password" },
       },
 
       async authorize(credentials, req) {
-        // TODO: confused about where exactly things need to be encapsulated regarding server/client side logic asnd this file and the button
-        // that triggers the signIn process.
+        console.log(credentials);
         await dbConnect();
         if (credentials?.password! || credentials?.username! === null)
```

Comment:
```suggestion
        if (credentials?.password === null || credentials?.username! === null)
```

The "!" at the end of the variable doesn't make it a negative. That just asserts that is exists. You essentially had "if there's a password, throw an error"